### PR TITLE
scan hardcoded images for vulnerabilities

### PR DESCRIPTION
**Pull request by eatwithforks**
might break projects that use vulnerability scanner but have non-gcr sidecars/other containers, those can disable the flag on the stage to make their deploys work while they move everything to gcr.

flag can be disabled on the stage edit page.
ie. projects/truth_service/stages/master/edit
screen shot 2019-01-14 at 9 26 08 am

@zendesk/compute @zendesk/bre

### CODE REVIEW

**Comment by hgokavarapuz** [comment]
Can please update in the PR how to do it, it would be helpful.

**Review by grosser — APPROVED** [review]

File: plugins/gcloud/test/samson_gcloud/samson_plugin_test.rb [hunk 1]
```diff
@@ -111,6 +111,43 @@ def expects_sleep(times)
...
     end
+
+    describe "gcloud vulnerabilty scanning" do
+      with_env GCLOUD_IMAGE_SCANNER: "true", GCLOUD_ACCOUNT: 'acc', GCLOUD_PROJECT: 'example'
+
+      def fire
+        Samson::Hooks.fire(:ensure_docker_image_has_no_vulnerabilities, stage, image)
+      end
+
+      let(:image) { +'foo.com/example/bar' }
```
Comment (hgokavarapuz) [inline plugins/gcloud/test/samson_gcloud/samson_plugin_test.rb]: what is that "+" here ? Is that intended ?

**Review by hgokavarapuz — APPROVED** [review]
small nits i guess, other than that looks fine to me.

File: plugins/gcloud/test/samson_gcloud/samson_plugin_test.rb (hunk 1 above)
Comment (eatwithforks) [reply plugins/gcloud/test/samson_gcloud/samson_plugin_test.rb]: to make variable nonfrozen. because `# frozen_string_literal: true` is at top of file.

**Pull request merged**